modules/cudf/src/data_frame.ts

```diff
@@ -70,6 +70,93 @@ function _invokeIfNumericSeries<P extends keyof T, T extends TypeMap, R extends
   return Series.new(series._col as Column<R>);
 }
 
+function _concat<DFs extends DataFrame[], T extends TypeMap>(...dfs: DFs): DataFrame {
+  const all_column_names = [...dfs
+                              .reduce((namesMap, df) => df.names.reduce(
+                                        (namesMap, name) => namesMap.set(name, true), namesMap),
+                                      new Map<string, boolean>())
+                              .keys()];
+
+  /**
+   * Array<Array<(Column|null)>> -- If a DF has a Column for a given name, it will be in the list
+   * otherwise there will be a null in that slot. For example:
+   * ```
+   * concat(new DataFrame({a, b}), new DataFrame({b, c}))
+   *
+   * columnsPerDF == [
+   *  [dfs[0].get("a"), dfs[0].get("b"),            null],
+   *  [           null, dfs[1].get("b"), dfs[1].get("c")]
+   * ]
+   * ```
+   */
+  const columns_per_df: any[][] =
+    dfs.map((df) => all_column_names.map((name) => df.has(name) ? df.get(name)._col : null));
+
```

Comment:
What do we want to do if `columns_per_df` is empty? Throw? return early?

Reply:
I haven't checked cudf but pandas raises an exception in this case.